dvc/repo/add.py

```diff
@@ -0,0 +1,43 @@
+import os
+
+
+def add(self, fname, recursive=False):
```

Comment:
The idea of extracting this code from the class is really great! No large classes anymore 🎉

The proposed design might be improved a bit. It has broken modularization issue. Now we have `command/add.py` as well as `dvc/repro/add.py` (for each of the commands). Both of the files have to be changed if a new cmd param was added.

I might miss something but I don't see any issues with moving all the extracted methods to `CmdX` classes. For the add command it will look like:
```
class CmdAdd(CmdBase):
    def add(self, repro, fname, recursive=False): # <-- moved from repro/add.py
        ....

    def run(self):
        for target in self.args.targets:
            try:
                # self.repo.add(target, recursive=self.args.recursive)
                #!!! Indirect call of repro.add() from ^^ should be replacted to direct call:
                self.add(self.repro, target, recursive=self.args.recursive)
            except DvcException:
                logger.error("failed to add file")
                return 1
```

Reply:
@dmpetrov `dvc/repo` is about the python API(that is internal for now, but we have plans to make it public in the future) and `dvc/command` is a CLI, that just wraps around it.  It seems suitable to keep those two separated.

Reply:
Then it makes sense to "duplicate" params code in cmd level and API level. Thank you!

Class `Repro` is still kind of large - the methods were extracted to separate files but it is still a single class. Importing methods looks a bit strange:
```
class Repo(object):
    from dvc.repo.destroy import destroy
    from dvc.repo.install import install
```

Does it make sense to create repo classes instead of repo methods? For API purposes these classes can be easily represented by some facade class to make API smooth.